precompiles/staking/method_distribute.go

```diff
@@ -0,0 +1,116 @@
+package staking
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+	fungibletypes "github.com/zeta-chain/node/x/fungible/types"
+)
+
+var (
+	zrc20lockerAddress = common.HexToAddress("0x0000000000000000000000000000000000000067")
+)
+
+// function distribute(address zrc20, uint256 amount) external returns (bool success)
+func (c *Contract) distribute(
+	ctx sdk.Context,
+	evm *vm.EVM,
+	contract *vm.Contract,
+	method *abi.Method,
```

Comment:
Do we need this arg? I think we can handle it outside of this code
```suggestion

```

Reply:
The main reason for this method to be passed as a reference is to get an ABI-packed result in `return method.Outputs.Pack(true)`.
I think it's cheaper just to pass a reference instead of doing something like `return c.Abi().Methods[Distribute].Pack(true).